lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -2129,8 +2129,15 @@ namespace Js
                 {
                     JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedFunction, _u("construct"));
                 }
-                newThisObject = JavascriptOperators::NewScObjectNoCtor(targetObj, scriptContext);
-                args.Values[0] = newThisObject;
+                if (!isCtorSuperCall)
+                {
+                    newThisObject = JavascriptOperators::NewScObjectNoCtor(targetObj, scriptContext);
+                    args.Values[0] = newThisObject;
+                }
+                else
```

Comment:
I believe we don't need this else clause at all since `newThisObject` is not used again below. We only need to not overwrite `args.Values[0]`. The below loop is going to copy all the existing arguments into the `newValues` array anyway. In fact, it doesn't appear we even need to declare `newThisObject`. The above line could just say:

```c++
args.Values[0] = JavascriptOperators::NewScObjectNoCtor(targetObj, scriptContext);
```

Reply:
It's used on line 2187: `aReturnValue = newThisObject;`

---
In reply to: [178616915](https://github.com/Microsoft/ChakraCore/pull/4908#discussion_r178616915) [](ancestors = 178616915)

Reply:
Oh thanks for pointing this out. I searched for this name but it didn't highlight that one on line 2187. I see no problems with this code, then.

Reply:
Could in theory replace aReturnValue below with args.Values[0]. If you want to remove a name from the code.